### DISCUSSION TITLE
What can this do that tmuxinator can't?

### DISCUSSION THREAD

**Discussion by OdinsHat** [tmux-python/tmuxp#468] [Show and tell]
**Simple question:** I already know [tmuxinator](https://github.com/tmuxinator/tmuxinator) as do a lot of people. Does this have any benefits over that? What are they?

**Comment by tony** [comment]
I haven't used tmuxinator in years.  (Though I used to, as well as teamocil, before tmuxp)

I used to keep a comparison at http://tmuxp.git-pull.com/en/latest/about.html#differences-from-tmuxinator-teamocil but it could use an update. If you (or anyone else) likes tmuxp feel free to update the comparison.

**Comment by OdinsHat** [comment]
Before I send you an email offering to help I'll do an update of this article and send it to you, Tony. Better than the all-too-common "volunteer-disappear" cycle. Hopefully sort Sat or Sun (24th Feb) and send to you in the same RST format you have it in. I'll be editing some existing bits to improve readability - don't take offence. :wink:

You'd be surprised how much benefit comparison pages have on getting potential users interested. [slant.co](http://slant.co) have built an entire business model on it!

**Anyone looking at this - I'm tackling it for now.**

**Comment by npearson72** [comment]
I'll tell you why I switched to tmuxp. I usually work on multiple Rails projects; some of which have substantially different versions of Ruby. I use a Ruby manager and tmux to switch between these different environments. Since tmuxinator uses Ruby, there were cases where some versions of the Ruby I was using resulted in deprecation warnings.

I hacked a solution together that used a global Ruby version just for tmuxinator, so that I could switch between my tmux sessions, but ultimately I wasn't keen on this approach.

Tmuxp came to the rescue. Just wish it didn't leave screen artifacts when switching to complex layouts:

https://github.com/tmux-python/tmuxp/issues/564
https://github.com/tmux-python/tmuxp/issues/365

**Comment by tony** [comment]
@npearson72 Thank you for that.

I converted this to a discussion! Let's see how this goes!
